users/spidey3/spidey3.c

```diff
@@ -0,0 +1,87 @@
+#include QMK_KEYBOARD_H
+
+#include <spidey3.h>
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+// If console is enabled, it will print the matrix position and status of each key pressed
+// #ifdef CONSOLE_ENABLE
+//   if (debug_enable) {
+//     uprintf("KL: kc: %u, col: %u, row: %u, pressed: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed);
+//   }
+// #endif
+
+  if (record->event.pressed) {
+    switch (keycode) {
+#ifndef NO_DEBUG
+      // Re-implement this here, but fix the persistence!
+      case DEBUG:
+        debug_enable ^= 1;
+        if (debug_enable) {
+          print("DEBUG: enabled.\n");
+        } else {
+          print("DEBUG: disabled.\n");
+        }
+        eeconfig_update_debug(debug_config.raw);
+#endif
+        return false;
+      case SPI_LNX:
+#ifdef CONSOLE_ENABLE
+	if (debug_enable) { print("SPIDEY3: SPI_LNX\n"); }
+#endif
+        layer_off(_OSX);
+        eeconfig_update_default_layer(_BASE);
+        default_layer_set(_BASE);
```

Comment:
https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

```suggestion
       set_single_persistent_default_layer(_BASE);
```

Reply:
The problem with just running set_single_persistent_default_layer() is that it doesn't seem to trigger a call to layer_state_set_user(), which is a problem, because that is where the rgb setting happens.

Reply:
Aha! I just noticed that there is a separate mixin - default_layer_state_set_user(). I've refactored to use that.